### PR TITLE
[OSDOCS-3592]: Note for Hyper-V gen2 images in machine set YAML

### PR DESCRIPTION
**Pull request by jeana-redhat**
Version(s):
4.11+

Issue:
[OSDOCS-3592](https://issues.redhat.com//browse/OSDOCS-3592) for [CORS-1916](https://issues.redhat.com//browse/CORS-1916)

Link to docs preview:
- [Sample YAML for a machine set custom resource on Azure](http://file.rdu.redhat.com/jrouth/OSDOCS-3592-hypervgeneration-v2-azure/machine_management/creating_machinesets/creating-machineset-azure.html#machineset-yaml-azure_creating-machineset-azure)
- [Sample YAML for a machine set custom resource on Azure](http://file.rdu.redhat.com/jrouth/OSDOCS-3592-hypervgeneration-v2-azure/machine_management/creating-infrastructure-machinesets.html#machineset-yaml-azure_creating-infrastructure-machinesets) (infra)

Additional information:
This is just the machine set component of this effort. Moved the `taints` parameter to the end of the YAML to make the conditionals less horrifying (_and_ it's alphabetical :nerd_face:).

### CODE REVIEW

File: modules/machineset-yaml-azure.adoc [hunk 1]
```diff
@@ -151,18 +143,17 @@ $  oc -n openshift-machine-api \
...
-<5> Specify the region to place machines on.
-<6> Specify the zone within your region to place machines on. Be sure that your region supports the zone that you specify.
 endif::infra[]
 ifdef::infra[]
 <2> Specify the `<infra>` node label.
 <3> Specify the infrastructure ID, `<infra>` node label, and region.
+endif::infra[]
 <4> Optional: Specify the machine set name to enable the use of availability sets. This setting only applies to new compute machines.
-<5> Specify a taint to prevent user workloads from being scheduled on infra nodes.
+<5> Specify an image that is compatible with your instance type. If your instance type is only compatible with Hyper-V generation V2 images, you must use a Hyper-V generation V2 image and append `-gen2` to the `resourceID` value.
```
Comment (patrickdillon) [inline modules/machineset-yaml-azure.adoc:152]: I am happy with this and think it is clear enough. 

If we are telling users to edit existing machinesets, it is possible-even probable-the image id would be a gen2 image already. So perhaps we want to say something like:

```suggestion
<5> Specify an image that is compatible with your instance type. The Hyper-V generation V2 images created by the installer have a `-gen2` suffix, while V1 images have the same name without the suffix.
```

Comment (jeana-redhat) [reply modules/machineset-yaml-azure.adoc:152]: Aha - that is very helpful, and I think will help users better than what I have here. Thanks!

**Comment by jeana-redhat** [comment]
@MayXuQQ PTAL when you can

File: modules/machineset-yaml-azure.adoc [hunk 2]
```diff
@@ -151,18 +143,17 @@ $  oc -n openshift-machine-api \
...
-<5> Specify the region to place machines on.
-<6> Specify the zone within your region to place machines on. Be sure that your region supports the zone that you specify.
 endif::infra[]
 ifdef::infra[]
 <2> Specify the `<infra>` node label.
 <3> Specify the infrastructure ID, `<infra>` node label, and region.
+endif::infra[]
 <4> Optional: Specify the machine set name to enable the use of availability sets. This setting only applies to new compute machines.
-<5> Specify a taint to prevent user workloads from being scheduled on infra nodes.
+<5> Specify an image that is compatible with your instance type. The Hyper-V generation V2 images created by the installer have a `-gen2` suffix, while V1 images have the same name without the suffix.
```
Comment (tmalove) [inline modules/machineset-yaml-azure.adoc:152]:
```suggestion
<5> Specify an image that is compatible with your instance type. The Hyper-V generation V2 images created by the installation program have a `-gen2` suffix, while V1 images have the same name without the suffix.
```

**Comment by tmalove** [comment]
Hi @jeana-redhat I left a suggestion, per https://redhat-documentation.github.io/supplementary-style-guide/#_t
Everything else LGTM.

File: modules/machineset-yaml-azure.adoc (hunk 2 above)
Comment (jeana-redhat) [reply modules/machineset-yaml-azure.adoc:152]: hm good question - @mjpytlak what's the norm for our docs on this term?

Comment (mjpytlak) [reply modules/machineset-yaml-azure.adoc:152]: `Installer` is all over the install doc set, but it should be `installation program`. I typically fix these type of instances as I go.

Comment (mjpytlak) [reply modules/machineset-yaml-azure.adoc:152]: > hm good question - @mjpytlak what's the norm for our docs on this term?

`Installer` is all over the install doc set, but it should be` installation program`. I typically fix these type of instances as I go.

**Comment by patrickdillon** [comment]
LGTM

**Comment by kalexand-rh** [comment]
LGTM

File: modules/machineset-yaml-azure.adoc (hunk 2 above)
Comment (jeana-redhat) [reply modules/machineset-yaml-azure.adoc:152]: Thanks! I'll make sure this one is correct from the get-go then :slightly_smiling_face:

**Comment by MayXuQQ** [comment]
compared with the created yaml by  versioned-installer-disk-encryption-set-ci, whether the difference is acceptable ? @patrickdillon 
 https://docs.google.com/document/d/1mGdNeQ0g9wA9vw3A2QOQ27fzXaGUOBaeKxHOyNMDlNE/edit?usp=sharing

**Comment by MayXuQQ** [comment]
compared with the created yaml by  versioned-installer-disk-encryption-set-ci, whether the difference is acceptable ? @patrickdillon 
 https://docs.google.com/document/d/1mGdNeQ0g9wA9vw3A2QOQ27fzXaGUOBaeKxHOyNMDlNE/edit?usp=sharing

**Comment by patrickdillon** [comment]
> compared with the created yaml by versioned-installer-disk-encryption-set-ci, whether the difference is acceptable ? @patrickdillon https://docs.google.com/document/d/1mGdNeQ0g9wA9vw3A2QOQ27fzXaGUOBaeKxHOyNMDlNE/edit?usp=sharing

I think those differences are fine. I would not be concerned about those.

**Comment by jeana-redhat** [comment]
Thanks @MayXuQQ and @patrickdillon for the very fast replies - can I consider this validated in that case?

**Comment by jeana-redhat** [comment]
Force pushed to force rebuild before merging, no change.

**Pull request merged**